src/Core/Types/Unifier.cs

```diff
@@ -534,6 +535,17 @@ public DataType UnifyStructures(StructureType a, StructureType b)
 			return mem;
 		}
 
+        private string MakeFieldName(StructureField fa, StructureField fb)
+        {
+            if (fa.NameSet && fb.NameSet && fa.Name != fb.Name)
+                throw new NotSupportedException("Both structure fields have names");
+            if (fa.NameSet)
```

Comment:
It would be nice if there were more context in the exception so that trouble shooting this is easier. For instance, change this to 'bool TryMakeFieldName(fa, fb, out string name)`. Have it return `false` if it fails because of name conflict. Then make the caller of `TryMakefieldName` throw an exception to the effect of "Failed to unify field 'xxxx' in stucture 'yyyy' with field 'zzzz' in structure 'www'."